specs/governance/alligator.md

```diff
@@ -0,0 +1,114 @@
+# Alligator
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Integration with `GovernanceToken`](#integration-with-governancetoken)
+- [Governor](#governor)
+- [Subdelegations](#subdelegations)
+  - [Subdelegation Rules](#subdelegation-rules)
+- [`afterTokenTransfer`](#aftertokentransfer)
+- [View functions](#view-functions)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `Alligator` contract implements advanced delegation features for the `GovernanceToken` contract. It is based on
+the Alligator V2 contract, with the following modifications from AlligatorOP:
+
+- uses hashed proxy rules to reduce calldata size
+
+- assumes one proxy per owner
+
+- uses proxies without deploying them
+
+- casts votes in batch directly to governor
+
+- adds alternative methods to limit the sender's voting power when casting votes
+
+- implements upgradeable version of the contract
+
+- adds method for casting multiple by signature
+
+## Integration with `GovernanceToken`
+
+The `Alligator` contract integrates with the `GovernanceToken` contract through a hook-based approach. The
+`GovernanceToken` contract calls the `Alligator` contract's `afterTokenTransfer` function after a token transfer. This
+allows the `Alligator` contract to consume the hooks and update its delegation and checkpoint mappings accordingly.
+If either of the addresses passed as arguments to `afterTokenTransfer`, `from` and `to`, have not been migrated,
+`Alligator` copies the address's delegation and checkpoint data from the `GovernanceToken` contract to its own state.
+
+## Governor
+
+The `Alligator` contract MUST provide a method for validating subdelegation rules and casting a vote on the governor.
+It should also provide methods for variants such as for
+
+- casting votes with reason
```

Comment:
How does voting work?

Reply:
updated content, but voting remains the same/unaffected (done via the Governor contract)